src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/instance_blackrock_spire.cpp

```diff
@@ -168,7 +170,8 @@ class instance_blackrock_spire : public InstanceMapScript
                         creature->DisappearAndDie();
                     break;
                 case NPC_FINKLE_EINHORN:
-                    creature->AI()->Talk(SAY_FINKLE_GANG);
+                    if (GetBossState(DATA_THE_BEAST) == DONE && loot == LOOT_SKINNING)
```

Comment:
Loot is being used here unintialized, so this condition is never true, because loot is never LOOT_SKINNING. Since the crash happens in the line below, this apparently seems to "work" because the line that causes it to crash never happens now - the npc never says anything

```cpp
if (creature->AI())
{
    creature->AI()->Talk(XXXXX);
}
```

But it may incur the same issue if the AI is not initialized in time.